pkg/buildplan/artifact_test.go

```diff
@@ -54,14 +54,13 @@ func TestArtifact_Dependencies(t *testing.T) {
 			want: []string{
 				"00000000-0000-0000-0000-000000000002",
 				"00000000-0000-0000-0000-000000000003",
-				"00000000-0000-0000-0000-000000000001",
```

Comment:
The old assertion failed due to my fix here:

https://github.com/ActiveState/cli/pull/3467/files#diff-9a4a1ec82a2498e8c04daf3594acd7e9f1de41731d7ef7539864881784a5c5c9R270

Before that you would get artifacts that had technically already been "seen". ie. asking for deps on artifact ID `1` would return `1`. Which is not expected (and ironically; also a cycle).